Feature/proximidade entre palavras

gui-kratos007 opened the pull request.

gui-kratos007 committed:
Criei um menu interativo com as funções que ja tinha criado antes, que permite ao usuário ecolher qual função ele deseja utilizar.

gui-kratos007 committed:
Criei um tratamento de erro para quando o usuário digitar um valor diferente das opções válidas, seja escrevendo um número inteiro que não corresponda às opções, seja escrevendo alguma coisa diferente do número inteiro. Além disso acrescentei mensagens de "Programa finalizado" para aparecer quando o usuário escolher tal opção.

gui-kratos007 committed:
Criei o arquivo close_words.py, e inseri nele alguns códigos para tentar iniciar, mas por enquanto nenhum completo.

gui-kratos007 committed:
Atualizei o código e tentei fazer a função frequency, mas ainda está incompleta

gui-kratos007 committed:
Criei a função desempate, que escolhe um item aleatório de uma lista de itens, mas faltam alguns ajustes para ser realmente aleatório.Além disso, atualizei a função frequency, mas ainda está faltando poucos ajustes para ela funcionar corretamente.

gui-kratos007 committed:
Criei as funções print_percentage e print_most_commons, e atualizei a frequency, retirando muita coisa desnecessária e substituindo outras coisas por funções que criei. Além disso já atualizei o arquivo menu.py com as novas funções para adicionar a opção "buscar palavras proximas mais frequentes" no menu.

gui-kratos007 committed:
Consertei o erro que estava na função "print_most_commons" e acertei algumas coisas do menu. Assim finalizei essa feature.

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+import random
+
+def get_text(path):
```
Adicione documentação em cada metodo ou função, exemplo como f unciona.. verifique sobre como gerar docstrings de uma codigo python.

```
def get_text(path):
    """
    Descrição da função
    "'"
```

diegopasti replied:
Fazer isso em todos os métodos

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+    print('-' * 150)
+    print("Porcentagem da frequencia dos anteriores: \n")
+    for item in dict1:
+        percentage_value = (dict1.get(item, 0) / total_itens) * 100
```
Evita fazer muitas operações por linha.. crie variaveis pra cada operação que facilita o reconhecimento da lógica por outras pessoas.

`percentage_value = (dict1.get(item, 0) / total_itens) * 100`

pode ficar assim:

`
ocurrences = dict1.get(item, 0)
percentage_value = (ocurrences / total_itens) * 100`

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+    print('-' * 150)
+    print("Porcentagem da frequencia dos posteriores: \n")
+    for item in dict2:
+        percentage_value = (dict2.get(item, 0) / total_itens2) * 100
```
cria uma função utilitária que calcula isso em vez de definir a mesma conta com a outra lista

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+        immediate_previous = desempatar(lista1)
+        immediate_subsequent = desempatar(lista2)
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) > 1 >= len(lista2):
```
len(lista1) > 1 >= len(lista2) 

isso aqui funciona?

diegopasti replied:
ta com cara que nao kk

gui-kratos007 replied:
Quando eu testei aqui funcionou. Quando fiz a primeira coloquei um and no meio das duas lens, mas o proprio pycharm me recomendou escrever desse jeito.

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+        immediate_subsequent = lista2[0]
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) <= 1 < len(lista2):
+        immediate_previous = lista1[0]
```
3 blocos de código fazendo algo muito parecido, que é definindo quem é o imediate_previous e subsequente, deve ahaver meio de simplificar esse código.. muitas condições podem indicar ponto de melhoria (se for aninhadas então..)

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+        return f"{lista1[0]} {word} {lista2[0]}"
+
+
+def frequency(txt):
```
função com mais de 15 linhas pode ser considerada grande.. se uma função for bem definida mesmo e fizer só o que deve, ela não usa muitas linhas.. um função muito longa é mais dificil de manter que varias pequenas (pois vc separada a lógica nomeando cada parte, deixa documentação e a depuração é mais direta)

diegopasti commented on close_words.py:
```diff
@@ -0,0 +1,181 @@
+        # print(subsequent_word)
+
+        # Verifica se há empate de palavras mais frequentes
+        previous_word_tie = [word for word in previous if previous[word] == previous[previous_word]]
```
entende o  que está fazendo aqui e quais as vantagens dessa abordagem?

gui-kratos007 replied:
Sim, ele coloca uma word na previous_word_tie, para cada word que estiver no dict previous e se a previous[word], ou seja, a palvra no indice que estiver sendo percorrido em previous, for igual a previous[previous_word], que é o índice da palavra que aparece mais vezes no dict previous

gui-kratos007 replied:
A vantagem é que ele forma uma lista com todas as condições necessárias em apenas uma linha

diegopasti commented on menu.py:
```diff
@@ -0,0 +1,247 @@
+        try:
+            answer = int(input("Digite sua opção: "))
+            if answer == 1:
+                path = input("Entre com o path do arquivo que deseja processar: ")
```
da pra criar uma função pra cada opção, em vez de deixar tudo aqui direto.. ai a lógica do menu não se mistura com a lógica da função escolhida.

diegopasti commented on menu.py:
```diff
@@ -0,0 +1,247 @@
+                print(text)
+                print()
+            elif answer == 2:
+                path = input("Entre com o path do arquivo que deseja processar: ")
```
fazer uma função pra cada ação que vai  ser executada..nos demais casos tbm

diegopasti commented on menu.py:
```diff
@@ -0,0 +1,247 @@
+
+def print_most_commons(lista1, lista2, word):
+    if len(lista1) > 1 and len(lista2) > 1:
+        immediate_previous = desempatar(lista1)
```
os proximos blocos fazem a mesma coisa, só que com valores diferentes.. 

deve dar pra criar um dicionario pra controlar as variações e ai dado a operação lógicaentregar os valores direto

gui-kratos007 committed:
Consertei um erro do menu.py e no close_words.py, e modifiquei outras partes dele que estavam de um jeito pior do que agora.

gui-kratos007 committed:
Coloquei as descrições nas funções e criei novas funções para diminuir a frequency que estava muito longa

gui-kratos007 committed:
Coloquei as descrições nas funções e criei novas funções para diminuir a frequency que estava muito longa

gui-kratos007 committed:
Coloquei uma função para a lógica da primeira opção do menu

gui-kratos007 committed:
Acertei a maioria das coisas pedidas no github, como diminuir tamanho de algumas funções, criando outras se necessário;
colocar a lógica e as ações das opções do menu em funções;
etc
mas ainda falta uma das coisas pedidas para resolver

gui-kratos007 committed:
Resolvi a última coisa que faltava das pedidas no github, mas ainda tem poucos detalhes que quero acertar, portanto, o que foi pedido está feito, mas acho que ainda tenho que aperfeiçoar o código em alguns aspectos.

gui-kratos007 committed:
Nessa versão terminei o último pedido de revisão e fiz um dicionário para os valores possíveis na função "print_most_commons", e ajustei algumas coisas no menu para faze-la se encaixar corretamente.
